stats/opentelemetry/client_metrics.go

```diff
@@ -0,0 +1,262 @@
+		if err != nil {
+			logger.Errorf("failed to register metric \"grpc.client.call.duration\", will not record")
+		} else {
+			registeredMetrics.clientCallDuration = ccs
+		}
+	}
+	csh.registeredMetrics = registeredMetrics
+}
+
+func (csh *clientStatsHandler) unaryInterceptor(ctx context.Context, method string, req, reply any, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+	ci := &callInfo{
+		target: csh.determineTarget(cc),
+		method: csh.determineMethod(method, opts...),
+	}
+	ctx = setCallInfo(ctx, ci)
+
+	startTime := time.Now()
+	err := invoker(ctx, method, req, reply, cc, opts...)
+	csh.perCallMetrics(ctx, err, startTime, ci)
+	return err
+}
+
+// determineTarget determines the target to record attributes with. This will be
+// "other" if target filter is set and specifies, the target name as is
+// otherwise.
+func (csh *clientStatsHandler) determineTarget(cc *grpc.ClientConn) string {
+	target := cc.CanonicalTarget()
+	if csh.mo.TargetAttributeFilter != nil {
+		if !csh.mo.TargetAttributeFilter(target) {
+			target = "other"
+		}
+	}
+	return target
+}
+
+// determineMethod determines the method to record attributes with. This will be
+// "other" if StaticMethod isn't specified or if method filter is set and
+// specifies, the method name as is otherwise.
+func (csh *clientStatsHandler) determineMethod(method string, opts ...grpc.CallOption) string {
+	if csh.mo.MethodAttributeFilter != nil {
+		if !csh.mo.MethodAttributeFilter(method) {
+			return "other"
+		}
+	}
+	for _, opt := range opts {
+		if _, ok := opt.(grpc.StaticMethodCallOption); ok {
+			return method
+		}
+	}
+	return "other"
+}
+
+func (csh *clientStatsHandler) streamInterceptor(ctx context.Context, desc *grpc.StreamDesc, cc *grpc.ClientConn, method string, streamer grpc.Streamer, opts ...grpc.CallOption) (grpc.ClientStream, error) {
+	ci := &callInfo{
+		target: csh.determineTarget(cc),
+		method: csh.determineMethod(method, opts...),
+	}
+	ctx = setCallInfo(ctx, ci)
+	startTime := time.Now()
+
+	callback := func(err error) {
+		csh.perCallMetrics(ctx, err, startTime, ci)
+	}
+	opts = append([]grpc.CallOption{grpc.OnFinish(callback)}, opts...)
+	s, err := streamer(ctx, desc, cc, method, opts...)
+	if err != nil {
+		return nil, err
+	}
+	return s, nil
+}
+
+func (csh *clientStatsHandler) perCallMetrics(ctx context.Context, err error, startTime time.Time, ci *callInfo) {
+	s := status.Convert(err)
+	callLatency := float64(time.Since(startTime)) / float64(time.Second)
+	if csh.registeredMetrics.clientCallDuration != nil {
+		csh.registeredMetrics.clientCallDuration.Record(ctx, callLatency, metric.WithAttributes(attribute.String("grpc.method", removeLeadingSlash(ci.method)), attribute.String("grpc.target", ci.target), attribute.String("grpc.status", canonicalString(s.Code())))) // needs method target and status should I persist this?
```

Comment:
question?

Reply:
Whoops, sorry, I don't know how I didn't catch this when I cleaned this up. Deleted.